engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -213,6 +222,15 @@ private void tryMigrateElementInstance(
                         .setProcessDefinitionKey(processDefinition.getKey())
                         .setBpmnProcessId(processDefinition.getBpmnProcessId())
                         .setTenantId(elementInstance.getValue().getTenantId())));
+
+    final ExecutableCatchEventSupplier targetElement =
+        processDefinition
+            .getProcess()
+            .getElementById(targetElementId, ExecutableCatchEventSupplier.class);
+    final BpmnElementContextImpl bpmnElementContext = new BpmnElementContextImpl();
+    bpmnElementContext.init(
+        elementInstance.getKey(), elementInstance.getValue(), elementInstance.getState());
```

Comment:
@korthout I still think that we should introduce the overloaded method for clarity of "only element instance key is used to unsubscribe from events", instead of passing the whole object. WDYT?